Tabbing, select all in cells, and change single click to ctrl click

cfonseca88 opened the pull request:
Thanks for this great npm for quill version 1. This module seems to work better than most of the Quill 2.0 table modules even, especially with the multi-line in cell!

I have added some minor optional key bindings which seem to fit nicely with what you have currently in keyboardHandler. I have added a tabbing bind so the user can quickly enter data move to the next cell or previous cell.

I have also added a custom optional select all for the cell alone, for quick delete, or edit of text. This works nicely with your implementation of multi-line per cell which has been appreciated.

Lastly, I added a different approach for cell select (previous single click). I appreciated that it was a separate function for click and double click but found it much easier as a ctrl click for selecting the cell. Regular click as the entry to cell to begin typing, etc.

cfonseca88 committed:
Tabbing, select all in cells, and change single click to ctrl click

Thanks for this great npm for quill version 1. This module seems to work better than most of the Quill 2.0 table modules even, especially with the multi-line in cell!

I have added some minor optional key bindings which seem to fit nicely with what you have currently in keyboardHandler. I have added a tabbing bind so the user can quickly enter data move to the next cell or previous cell.

I have also added a custom optional select all for the cell alone, for quick delete, or edit of text. This works nicely with your implementation of multi-line per cell which has been appreciated.

Lastly, I added a different approach for cell select (previous single click). I appreciated that it was a separate function for click and double click but found it much easier as a ctrl click for selecting the cell. Regular click as the entry to cell to begin typing, etc.

cfonseca88 committed:
Last commit build resulting files

cfonseca88 committed:
Shift tab tweak

When we are at the first cell and shift tab, go to electron default (not previous line etc). This makes it feel a little better!

dclement8 commented on src/demo.js:
```diff
@@ -59,6 +60,35 @@ const quill = new Quill(document.getElementById('quillContainer'), {
+            }
+          }
+        },
+        customShiftTab: {
```
```suggestion
        shiftTab: {
```

dclement8 commented on src/demo.js:
```diff
@@ -59,6 +60,35 @@ const quill = new Quill(document.getElementById('quillContainer'), {
+            return TableModule.keyboardHandler(this.quill, 'shiftTab', range, keycontext);
+          }
+        },
+        customSelectAll: {
```
```suggestion
        selectAll: {
```

dclement8 reviewed (commented):
Hi, thank you very much for your contribution!
I tested and everything works. I have made some suggestions on your PR.

Another suggestion: for cell selection, I think we should make it configurable:
- with the CTRL key (and single click for editing the cell) => your implementation
- single click (and double click for editing the cell) => previous implementation

https://quilljs.com/guides/building-a-custom-module/#using-options
CTRL key by default except if we pass "holdCtrlKeyForCellSelection: false" or "cellSelectionOnClick" option (or something similar, I'm not sure what the most appropriate name would be)

cfonseca88 committed:
clickOption suggestion and suggested keyboard bind change

cellSelectionOnClick set to true by default to match original demo unless otherwise optioned like below:

  "modules: {
    table: {
      cellSelectionOnClick:false
    },
}"

cfonseca88 commented:
Thanks for the suggestion! I went ahead and added that with your keybind suggestion aswell. I have made it so it is set to cellSelection on by default to match the original demo. If the user configs to table:true in modules it will be the default behavior aswell or unless otherwise optioned
